core/cache/model.go

```diff
@@ -480,6 +533,7 @@ func (m *Model) setDetails(details ModelChange) {
 		m.hub.Publish(modelConfigChange, hashCache)
 	}
 
+	m.updateSummary()
```

Comment:
I don't think this is O(N²), just O(N). A change to any machine or unit only requires updating the summary. We aren't triggering changes to all at once often, if ever. If we try to background any generation of this we lose the ability to synchronise over the calls. Also, determining when someone is asking is actually a problem in itself.

The generation of the summary object is very light weight. Is it the hash caclulation that concerns you? We could just store the string representation of the summary rather than a hash. The hash was used previously for quick comparisons, and that it was required for a string watcher. When writing this code it was more reuse of the pattern rather than necessity. Storing the string representation would suit the needs as much and not require the computation.